File: selfstudy/golang/temperatures_max.go
/*
task 242:
Temperatures - Max
You need to find the maximum value among daily temperatures in Celsius
*/

package main

import (
	"fmt"
	"reflect"
)

func HighestTemperatures(temps []int) int {
	if len(temps) == 0 { return 0 }
	m := temps[0]
	for _, v := range temps {
		if v > m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, HighestTemperatures([]int{10, 20, 30})},
		{[]int{50, 60, 70}, HighestTemperatures([]int{50, 60, 70})},
		{[]int{1, 2, 3}, HighestTemperatures([]int{1, 2, 3})},
		{[]int{100, 200}, HighestTemperatures([]int{100, 200})},
		{[]int{}, HighestTemperatures([]int{})},
	}
	
	for idx, t := range tests {
		res := HighestTemperatures(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/temperatures_max.js
/*
task 42:
Temperatures - Max
You need to find the maximum value among daily temperatures in Celsius
*/

function highest_temperatures(temps) {
  return temps.length ? Math.max(...temps) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: highest_temperatures([10, 20, 30]) },
  { input: [50, 60, 70], expected: highest_temperatures([50, 60, 70]) },
  { input: [1, 2, 3], expected: highest_temperatures([1, 2, 3]) },
  { input: [100, 200], expected: highest_temperatures([100, 200]) },
  { input: [], expected: highest_temperatures([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('highest_temperatures', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(highest_temperatures(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, highest_temperatures(t.input) === t.expected || JSON.stringify(highest_temperatures(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
